odoo/tools/config.py

```diff
@@ -543,8 +543,8 @@ def die(cond, msg):
         )
 
         self.options['init'] = opt.init and dict.fromkeys(opt.init.split(','), 1) or {}
-        self.options['demo'] = (dict(self.options['init'])
-                                if not self.options['without_demo'] else {})
+        self.options['without_demo'] = self.options['without_demo'] not in (False, '', 'False', 'false')
+        self.options['demo'] = (dict(self.options['init']) if self.options['without_demo'] else {})
```

Comment:
```suggestion
        self.options['demo'] = (dict(self.options['init']) if not self.options['without_demo'] else {})
```
Shouldn't the `not` still be in the `if` condition??